File: app/src/main/java/pl/rpieja/flat/dto/Charge.java
package pl.rpieja.flat.dto;

import org.jetbrains.annotations.Nullable;

import java.util.Date;
import java.util.List;

/**
 * Created by radix on 2017-11-01.
 */

public class Charge implements ChargeLike {
    public String name, rawAmount;
    public Date date;
    public int from, id;
    public List<User> to;
    public double amount;

    @Nullable
    @Override
    public Double getChargeAmount() {
        return amount;
    }

    @Nullable
    @Override
    public String getChargeName() {
        return name;
    }

    @Nullable
    @Override
    public List<User> getFromUsers() {
        return null;
    }

    @Nullable
    @Override
    public List<User> getToUsers() {
        return to;
    }
}

File: app/src/main/java/pl/rpieja/flat/dto/Charge.kt
package pl.rpieja.flat.dto

import android.annotation.SuppressLint
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.util.Date

@SuppressLint("ParcelCreator") // FIXME maybe caused by Kotlin Android ticket KT-19300
@Parcelize
data class Charge(
        var name: String,
        var rawAmount: String,
        var date: Date,
        var from: Int,
        var id: Int,
        var to: List<User>,
        var amount: Double
) : ChargeLike, Parcelable {
    override val chargeName: String
        get() = name
    override val chargeAmount: Double
        get() = amount
    override val fromUsers: List<User>
        get() = emptyList()
    override val toUsers: List<User>
        get() = to
}
